src/components/ResetPassword.js

```diff
@@ -0,0 +1,54 @@
+// ForgotPassword.js
+import React, { useState } from 'react';
+import { useHistory, useLocation } from 'react-router-dom';
+import { resetPassword } from '../apis/authApi'; // Make sure to import your API call function
+
+const ResetPassword = () => {
+  const history = useHistory();
+  const location = useLocation();
+  const searchParams = new URLSearchParams(location.search);
+  const token = searchParams.get('token');
+
+  const [password, setPassword] = useState('');
+  const [confirmPassword, setConfirmPassword] = useState('');
+  const [error, setError] = useState('');
+
+  const handleResetPassword = async () => {
+    try {
+      if (password !== confirmPassword) {
+        setError('Passwords do not match');
+        return;
+      }
+
+      await resetPassword({ token, password });
+
+      // Redirect to sign in page on success
+      history.push('/signin');
```

Comment:
Check how to do this with the `useNavigate` hook for react-router-dom v6

Reply:
Thank you , edited👍